Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -0,0 +1,281 @@
+
+#include <OptionParser.h>
+
+namespace DolphinTool
+{
+int ConvertCommand::Main(const std::vector<std::string>& args)
+{
+  auto parser = std::make_unique<optparse::OptionParser>();
+
+  parser->usage("usage: convert [options]... [FILE]...");
+
+  parser->add_option("-i", "--input")
+      .type("string")
+      .action("store")
+      .help("Path to disc image FILE.")
+      .metavar("FILE");
+
+  parser->add_option("-o", "--output")
+      .type("string")
+      .action("store")
+      .help("Path to the destination FILE.")
+      .metavar("FILE");
+
+  parser->add_option("-f", "--format")
+      .type("string")
+      .action("store")
+      .help("Container format to use. Default is RVZ. [%choices]")
+      .choices({"iso", "gcz", "wia", "rvz"});
+
+  parser->add_option("-s", "--scrub")
+      .action("store_true")
+      .help("Scrub junk data as part of conversion.");
+
+  parser->add_option("-b", "--block_size")
+      .type("int")
+      .action("store")
+      .help("Block size for GCZ/WIA/RVZ formats, as an integer. Suggested value for RVZ: 131072 "
+            "(128KiB)");
+
+  parser->add_option("-c", "--compression")
+      .type("string")
+      .action("store")
+      .help("Compression method to use when converting to WIA/RVZ. Suggested method: Zstandard "
+            "(zstd) "
+            "[%choices]")
+      .choices({"none", "zstd", "bzip", "lzma", "lzma2"});
+
+  parser->add_option("-l", "--compression_level")
+      .type("int")
+      .action("store")
+      .help("Level of compression for the selected method. Ignored if 'none'. Suggested value for "
+            "zstd: 5");
+
+  const optparse::Values& options = parser->parse_args(args);
+
+  // Validate options
+
+  // --input
+  const std::string input_file_path = static_cast<const char*>(options.get("input"));
+  if (input_file_path.empty())
+  {
+    std::cerr << "Error: No input set" << std::endl;
+    return 1;
+  }
+
+  // --output
+  const std::string output_file_path = static_cast<const char*>(options.get("output"));
+  if (output_file_path.empty())
+  {
+    std::cerr << "Error: No output set" << std::endl;
+    return 1;
+  }
+
+  // --format
+  const std::optional<DiscIO::BlobType> format_o =
+      ParseFormatString(static_cast<const char*>(options.get("format")));
+  if (!format_o.has_value())
+  {
+    std::cerr << "Error: No output format set" << std::endl;
+    return 1;
+  }
+  const DiscIO::BlobType format = format_o.value();
+
+  // Open the volume now for inspection
+  std::unique_ptr<DiscIO::Volume> volume = DiscIO::CreateVolume(input_file_path);
+  if (!volume)
+  {
+    std::cerr << "Error: Unable to open disc image" << std::endl;
+    return 1;
+  }
+
+  // --scrub
+  const bool scrub = static_cast<bool>(options.get("scrub"));
+
+  if (scrub && (format == DiscIO::BlobType::RVZ || volume->IsDatelDisc()))
+  {
+    std::cerr << "Error: Scrubbing an RVZ container or Datel disc is not supported";
+    return 1;
+  }
```

Comment:
Consider making the RVZ part just a warning. Scrubbing RVZ does work correctly, it's just that it saves almost no space. (Scrubbing Datel, on the other hand, most certainly doesn't work and should remain an error.)